packages/expect/src/__tests__/spyMatchers.test.ts

```diff
@@ -841,72 +992,108 @@ describe.each([
   'toReturnWith',
   'toHaveReturnedWith',
 ] as const)('%s', returnedWith => {
-  const caller = function (
-    callee: (...a: Array<unknown>) => void,
-    ...args: Array<unknown>
-  ) {
-    if (
+  function isNth(
+    returnedWith: string,
+  ): returnedWith is 'nthReturnedWith' | 'toHaveNthReturnedWith' {
+    return (
       returnedWith === 'nthReturnedWith' ||
       returnedWith === 'toHaveNthReturnedWith'
-    ) {
-      callee(1, ...args);
-    } else {
-      callee(...args);
-    }
-  };
+    );
+  }
 
   test('works only on spies or jest.fn', () => {
     const fn = function fn() {};
 
+    // @ts-expect-error: Testing runtime error
     expect(() => jestExpect(fn)[returnedWith]()).toThrowErrorMatchingSnapshot();
   });
 
   test('works when not called', () => {
     const fn = jest.fn();
-    caller(jestExpect(fn).not[returnedWith], 'foo');
 
-    expect(() =>
-      caller(jestExpect(fn)[returnedWith], 'foo'),
-    ).toThrowErrorMatchingSnapshot();
+    if (isNth(returnedWith)) {
+      jestExpect(fn).not[returnedWith](1, 'foo');
+
+      expect(() =>
+        jestExpect(fn)[returnedWith](1, 'foo'),
+      ).toThrowErrorMatchingSnapshot();
+    } else {
+      jestExpect(fn).not[returnedWith]('foo');
+
+      expect(() =>
+        jestExpect(fn)[returnedWith]('foo'),
+      ).toThrowErrorMatchingSnapshot();
+    }
   });
 
   test('works with no arguments', () => {
     const fn = jest.fn();
     fn();
-    caller(jestExpect(fn)[returnedWith]);
+
+    if (isNth(returnedWith)) {
+      // @ts-expect-error: TODO should types be fixed?
+      jestExpect(fn)[returnedWith](1);
+    } else {
+      // @ts-expect-error: TODO should types be fixed?
+      jestExpect(fn)[returnedWith]();
```

Comment:
Types require an argument to be passed to `returnedWith` matchers. Hm.. Should be fixed?

Reply:
See https://github.com/facebook/jest/pull/13385